### ISSUE TITLE
How to disable auto upgrading?

### ISSUE THREAD

**Issue by zomux** [zeit/hyper#1523]
- [o] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [o] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

- **OS version and name**: max os sierra
- **Hyper.app version**: 1.2.1
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: default one

I can not select text in tmux over ssh when updated to 1.2.0/1.2.1, but it is okay on 1.1.0.

Well, I just satisfy with the old version, how to disable auto upgrading?

**Comment by albinekb** [comment]
Don't know if it's possible, but a workaround is to clone the repo, checkout the version that works `1.1.0` then follow the instructions in the readme to build your own version...

**Comment by joshmanders** [comment]
I would like a way to disable automatic updates because I manage my updates via Homebrew Cask.

**Comment by hboomsma** [comment]
The feature might be nice, but it causes the terminal to become very slow on start up (minutes) over a unstable satellite link. Connecting to the internet on startup forces me to switch to another terminal emulator if the startup time is so long.

**Comment by mhall** [comment]
Especially with the issues in v3 that was just pushed out, disabling auto-upgrading (maybe with `updateChannel: none` or similar in the config file) is a must. Very disappointed to see this was closed as Won't Fix, and hope that can be reconsidered.

**Comment by theicfire** [comment]
Yup, v3 is broken for me and I can't disable auto update

**Comment by mhall** [comment]
I suspect this can be worked around by blackholing `releases.hyper.is` and `releases-canary.hyper.is` via Hosts file. I will test this and report back later today.

**Comment by pcnate** [comment]
can you manually launch the previous version after the update to avoid running the broken version? I was able to launch 2.1.2 in this manner

![image](https://user-images.githubusercontent.com/621940/57252887-f2fe8700-7012-11e9-8b76-64d782e0764e.png)

**Comment by MarxSoul55** [comment]
@mhall I have tried your method and can provide additional confirmation that it works! Thanks a bunch! Hopefully all these bugs with v3 will be fixed soon.

**Comment by aantron** [comment]
An option to disable automatic updating is really necessary. At least, even if there is an automatic check for updates, there should be the choice of whether to install or not. It's not reasonable to automatically install a **major** version upgrade for all users.

I had "worked around" issues in 3.0.0 by changing my shortcut to point back to 2.1.2. However, 2.1.2 has now installed 3.0.2 automatically, which has again edited the shortcut, and changed the config. This is a waste of time, as I don't need 3.x.x, and so would prefer to wait until issues are resolved, or even never upgrade.

**Comment by pcnate** [comment]
I have to agree. Some of us have to have a functional terminal and if it were to become broken and not be able to update we would have to switch to a development install which I don't think is documented (at least I haven't seen it). I know if I had a rough week getting it to function, I would just have to switch to something else.

A simple setting or even a flag for the shortcut would suffice.

**Comment by mhall** [comment]
@aantron @pcnate Obviously not as convenient as a setting in the config file (which I 100% agree needs to be included) but my [hosts file solution](https://github.com/zeit/hyper/issues/1523#issuecomment-489697965) does achieve what you're looking for and is probably easier than updating the shortcut with each upgrade.

**Comment by pcnate** [comment]
honestly I think auto-update is a good thing so long as some way to remediate a broken or problematic update is necessary. a shortcut based approach would be harder to find but easier to reset on an update. A setting would be easier to find and more permanent unless the update were to change that setting which would go against the notice in the settings file. Regardless, being notified that there is an update would be idea regardless of whether it occurs. On Ubuntu I noticed that it will provide you a link to go download (at least the .deb) approach does. A .app would probably just update and have no way to go back.

**Comment by Rideite** [comment]
Is this still relevant? Anyway I deleted Update.exe and that seems to have stopped auto update

**Comment by SebastianoF** [comment]
Still relevant for me. 
It would be great to have a third option in the config file `.hyper.js` as:
```

  config: {
       // choose either `'stable'` for receiving highly polished,
       // or `'canary'` for less polished but more frequent updates,
       // or `'no'` for not triggering update at startup.
       updateChannel: 'no',
       ...
```
